src/vellum/workflows/nodes/displayable/web_search_node/tests/test_node.py

```diff
@@ -0,0 +1,323 @@
+import pytest
+from unittest.mock import MagicMock
+
+import requests
+
+from vellum.workflows.errors.types import WorkflowErrorCode
+from vellum.workflows.exceptions import NodeException
+from vellum.workflows.inputs import BaseInputs
+from vellum.workflows.state import BaseState
+from vellum.workflows.state.base import StateMeta
+
+from ..node import WebSearchNode
+
+
+@pytest.fixture
+def mock_context():
+    """Create a mock context with vellum_client."""
+    context = MagicMock()
+    context.vellum_client._client_wrapper.get_headers.return_value = {"User-Agent": "vellum-python-sdk/1.0.0"}
+    return context
+
+
+@pytest.fixture
+def base_node_setup(mock_context):
+    """Basic node setup with required inputs."""
+
+    class Inputs(BaseInputs):
+        query: str
+        api_key: str
+        num_results: int
+
+    class State(BaseState):
+        pass
+
+    class TestableWebSearchNode(WebSearchNode):
+        query = Inputs.query
+        api_key = Inputs.api_key
+        num_results = Inputs.num_results
+
+    state = State(meta=StateMeta(workflow_inputs=Inputs(query="test query", api_key="test_api_key", num_results=3)))
+    node = TestableWebSearchNode(state=state)
+    node._context = mock_context
```

Comment:
[nit] let's pass in the context using `node = TestableWebSearchNode(state=state, context=mock_context)`